2024Robot_Team3484/src/main/include/subsystems/DrivetrainSubsystem.h

```diff
@@ -45,11 +45,17 @@ class DrivetrainSubsystem : public frc2::SubsystemBase {
         };
 
     private:
+    // Check if can be placed in constants
+            SC::SC_SwerveConfigs _swerve_front_left{10,11,20, 4.394};
+            SC::SC_SwerveConfigs _swerve_front_right{12,13,21,71.630};
+            SC::SC_SwerveConfigs _swerve_back_left{14,15,22,-26.103};
+            SC::SC_SwerveConfigs _swerve_back_right{16,17,23,-71.455};
```

Comment:
Yes, these can be moved to constants and passed through the constructor. The SwerveModule array below would be empty. You may have to change the datatype of the array to:
```c++
SwerveModule* _modules[4];
```
in the constructor you would then have:
```c++
DrivetrainSubsystem(SC::SwerveConfig* Cfg_Array)
{
    if (NULL != Cfg_Array)
    {
        for(int i = 0; i < 4; i++)
        {
            _modules[i] = new SwerveModule(Cfg_Array[i]);
        }
    }

    // Gyro initialization here

    // Odometry initialization here
}
```

and don't forget to null check the modules later on.